start_admin_sshd.sh

```diff
@@ -2,11 +2,20 @@
 
 set -e
 
+log() {
+  echo "$*" >&2
+}
+
 declare -r PERSISTENT_STORAGE_BASE_DIR="/.bottlerocket/host-containers/current"
 declare -r SSH_HOST_KEY_DIR="${PERSISTENT_STORAGE_BASE_DIR}/etc/ssh"
 declare -r USER_DATA="${PERSISTENT_STORAGE_BASE_DIR}/user-data"
 
-declare -r LOCAL_USER="ec2-user"
+# Fetch custom user from user-data json (if any). Default to ec2-user if null or failed.
+if ! LOCAL_USER=$(jq -e -r '.["user"] // "ec2-user"' "${USER_DATA}" 2>/dev/null); then
+  log "Failed to set user with user-data. Proceeding with 'ec2-user'."
+  declare -r LOCAL_USER="ec2-user"
+fi
```

Comment:
```suggestion
LOCAL_USER="$(jq -e -r '.["user"] // "ec2-user"' "${USER_DATA}" 2>/dev/null)"
if [ "${?}" -ne 0 ] || [[ ! ${LOCAL_USER} =~ ^[a-z_][a-z0-9_-]{0,31}$ ]]; then
  log "Failed to set user from user-data. Proceeding with 'ec2-user'."
  LOCAL_USER="ec2-user"
fi
```
The `// "ec2-user"` means that the only time we hit the conditional is from bad JSON syntax.  `//` is nice to prevent outputting false/null, but unfortunately it doesn't catch empty strings, so user data with `{"user": ""}` would set LOCAL_USER="" and cause later commands to fail.  So, I think we need the extra check.  And if we're adding the check, we might as well do a better one than -z, since invalid characters would also cause later commands to fail.  (I also suggested removing the `declare -r` because having the success case be writable and the failure case not seems odd?)